editor/plugins/material_editor_plugin.cpp

```diff
@@ -161,43 +162,68 @@ void MaterialEditor::edit(Ref<Material> p_material, const Ref<Environment> &p_en
 
 void MaterialEditor::_on_light_1_switch_pressed() {
 	light1->set_visible(light_1_switch->is_pressed());
+	EditorSettings::get_singleton()->set_project_metadata("inspector_options", "material_preview_light1", light1->is_visible());
 }
 
 void MaterialEditor::_on_light_2_switch_pressed() {
 	light2->set_visible(light_2_switch->is_pressed());
+	EditorSettings::get_singleton()->set_project_metadata("inspector_options", "material_preview_light2", light2->is_visible());
+}
+
+void MaterialEditor::_on_floor_switch_pressed() {
+	bool is_visible = !floor_instance->is_visible();
+	floor_instance->set_visible(is_visible);
+	floor_switch->set_pressed(is_visible);
+	probe->set_visible(is_visible);
+	EditorSettings::get_singleton()->set_project_metadata("inspector_options", "material_preview_floor", is_visible);
 }
 
 void MaterialEditor::_on_sphere_switch_pressed() {
+	if (shape == SPHERE) {
+		rot = Vector2(Math::deg_to_rad(-30.0), Math::deg_to_rad(0.0));
+	}
+	shape = SPHERE;
 	sphere_instance->show();
 	box_instance->hide();
 	quad_instance->hide();
-	box_switch->set_pressed(false);
-	quad_switch->set_pressed(false);
-	_set_rotation(-15.0, 30.0);
-	_store_rotation_metadata();
-	EditorSettings::get_singleton()->set_project_metadata("inspector_options", "material_preview_mesh", "sphere");
+	sphere_switch->set_pressed_no_signal(true);
+	box_switch->set_pressed_no_signal(false);
+	quad_switch->set_pressed_no_signal(false);
+	cam_zoom = 2.973804;
```

Comment:
Does this zoom value correspond to anything specific? There should be a comment above it, or it should use the value of a named constant.

Reply:
It comes from `visible_height / sin(vfov)` for precise camera framing.
I refactored the switch logic in the new push.